### PR TITLE
Register RadioGroupControl in React Vanilla

### LINKED ISSUE #1783: RadioGroupControl not registered in React Vanilla
### Describe the bug

Hai 

          I am unable to use Input type ="radio" and checkbox if config below schema and schema json with vanilla renders its displaying select button can you help me with that you people are already developed render for radio button and check box can you please share configuration for this types in vanilla renders  
          
          
          
          {
  "type": "object",
  "properties": {
    "exampleRadioEnum": {
      "type": "string",
      "enum": [
        "One",
        "Two",
        "Three"
      ]
    }
  }
}

{
  "type": "Control",
  "scope": "#/properties/exampleRadioEnum",
  "options": {
    "format": "radio"
  }
}

### Expected behavior

Radio button

### Screenshots

{
          "type": "Control",
          "scope": "#/properties/sex",
          "options": {
            "format": "radio"
          }
        },

"sex": {
      "title": "Sex",
      "type": "string",
      "enum": [
        "male",
        "female"
      ]
    },

image

### In which browser are you experiencing the issue?

google chrome latest 

### Framework

React

### RendererSet

Vanilla

### Additional context

dADSad

### PR DESCRIPTION
**Pull request by max-elia**
- Register RadioGroupControl and radioGroupControlTester
- Convert enum and oneOfEnum to same format
- Create Test for oneOfEnum
Closes #1783

Signed-off-by: Max Elia Schweigkofler <max_elia@hotmail.de>

### CODE REVIEW

**Comment by max-elia** [comment]
@LukasBoll

**Review by LukasBoll — APPROVED** [review]

File: packages/vanilla/src/controls/RadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      description,
+      this.state.isFocused,
+      appliedUiSchemaOptions.showUnfocusedDescription
+    );
+
+    const options: EnumOption[] =
+      (schema.enum?.map(enumToEnumOptionMapper)) ||
+      (schema.oneOf &&
+        (schema.oneOf as JsonSchema[]).map(oneOfToEnumOptionMapper)
+      );
```
Comment (sdirix) [inline packages/vanilla/src/controls/RadioGroup.tsx:78]: This manual computation is not necessary, we offer specific bindings for this, i.e. `withJsonFormsEnumControlProps` and `withJsonFormsOneOfEnumProps`. The bindings eventually call [mapStateToEnumControlProps](https://github.com/eclipsesource/jsonforms/blob/ba6bdc5a93044a872285cd71ed40d9a54f437561/packages/core/src/util/renderer.ts#L457-L470) or [mapStateToOneOfEnumControlProps](https://github.com/eclipsesource/jsonforms/blob/ba6bdc5a93044a872285cd71ed40d9a54f437561/packages/core/src/util/renderer.ts#L478-L490) which do this mapping for you and will hand them over as `options` prop.
```suggestion
```

File: packages/vanilla/src/controls/OneOfRadioGroupControl.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export const OneOfRadioGroupControl = (props: ControlProps & VanillaRendererProps) => {
+  return <RadioGroup {...props} />;
+};
+
+export const oneOfRadioGroupControlTester: RankedTester = rankWith(
+  3,
+  and(isOneOfEnumControl, optionIs('format', 'radio'))
+);
+
+export default withVanillaControlProps(withJsonFormsControlProps(OneOfRadioGroupControl));
```
Comment (sdirix) [inline packages/vanilla/src/controls/OneOfRadioGroupControl.tsx:48]:
```suggestion
export default withVanillaControlProps(withJsonFormsOneOfEnumProps(OneOfRadioGroupControl));
```

File: packages/vanilla/src/controls/RadioGroupControl.tsx [hunk 3]
```diff
@@ -24,91 +24,21 @@
...
-  }
-}
-
-export default withVanillaControlProps(
-  withJsonFormsControlProps(RadioGroupControl)
+export const radioGroupControlTester: RankedTester = rankWith(
+  3,
+  and(isEnumControl, optionIs('format', 'radio'))
 );
+export default withVanillaControlProps(withJsonFormsControlProps(RadioGroupControl));
```
Comment (sdirix) [inline packages/vanilla/src/controls/RadioGroupControl.tsx:44]:
```suggestion
export default withVanillaControlProps(withJsonFormsEnumProps(RadioGroupControl));
```

**Review by sdirix — CHANGES_REQUESTED** [review]
The code works, however the bindings should be constructed slightly differently. We'll let the core framework do the `options` calculation and bind correspondingly.

File: packages/vanilla/src/controls/RadioGroupControl.tsx (hunk 3 above)
Comment (max-elia) [reply packages/vanilla/src/controls/RadioGroupControl.tsx]: Thanks, implemented the changes.

File: packages/vanilla/test/renderers/RadioGroupControl.test.tsx [hunk 4]
```diff
@@ -79,12 +101,34 @@ describe('Radio group control', () => {
...
+        wrapper = mount(
+            <Provider store={store}>
+                <JsonFormsReduxContext>
+                    <OneOfRadioGroupControl schema={oneOfFixture.schema} uischema={oneOfFixture.uischema} />
+                </JsonFormsReduxContext>
+            </Provider>
+        );
+
+        const radioButtons = wrapper.find('input[type="radio"]');
+        //expect(radioButtons).toHaveLength(3);
```
Comment (sdirix) [inline packages/vanilla/test/renderers/RadioGroupControl.test.tsx:126]:
```suggestion
        expect(radioButtons).toHaveLength(3);
```

**Review by sdirix — APPROVED** [review]
Rebased on latest master. Works as expected!

**Pull request merged**